spec3 pipeline for WFSS modes

nden opened the pull request:
Change the calwebb_spec3 pipeline to allow processing of WFSS modes.

Also fixed a bug in extract_1d where the shape of the input data was not passed correctly to the extraction code in the case of a single 2D slit.

This was tested on one file from the regression tests - NIS-WFSS. More testing necessary.

Resolves #3415.

nden commented:
The problem in #3458 is also seen in the results of  testing this PR. The output name contains the string from `slit.name` instead of `slit.source_id` which is supposed to be integer. The name of the output of spec3 for NIRISS WFSS for a specific test is `jw87600017001_02101_00002_nis_sobject 10_x1d.fits`. Note that there's also a space in the name. This is fixed in extract_2d.

philhodge commented:
Should there also be a call to `combine_1d_spectra` for each source?

nden commented:
@philhodge Yes, good point, I missed that part of the ticket.
Is this unique to WFSS data?

hbushouse commented:
Yes, the need to call ``combine_1d`` is only for WFSS data.

hbushouse commented on CHANGES.rst:
```diff
@@ -33,6 +33,11 @@ extract_1d
+extract_2d
+----------
+
+- A space was removed from the names of grism objects in a MultiSlit model. [#3517]
```
Technically the white space was **replaced** by an underscore.

hbushouse commented on jwst/extract_2d/grisms.py:
```diff
@@ -383,7 +383,7 @@ def extract_grism_objects(input_model,
                 # The overall subarray offset is recorded in model.meta.subarray.
                 # nslit = obj.sid - 1  # catalog id starts at zero
-                new_slit.name = "Object {0}".format(obj.sid)
+                new_slit.name = "Object_{0}".format(obj.sid)
```
I wonder if we should make the name syntax here consistent with what we're doing for NRS MSA data, i.e. something that's just an integer without "Object_"?

nden replied:
I agree. 👍 for less typing

hbushouse commented on jwst/pipeline/calwebb_spec3.py:
```diff
@@ -20,6 +20,7 @@
 # Group exposure types
 MULTISOURCE_MODELS = ['MultiSlitModel']
 IFU_EXPTYPES = ['MIR_MRS', 'NRS_IFU']
+SLITLESS_TYPES = ['NIS_WFSS', 'NRC_WFSS'] #, 'NRC_TSGRISM']
```
NRC_TSGRISM is already comment out so it has no effect on anything, but we really could just completely remove it, because time-series grism data should never go through this pipeline anyway. They go through ``calwebb_tso3``.

nden commented:
I addressed all comments.
There's no data to test this code on. There's one NIRISS data file but the data looks like noise and although extract_1d does not crash the wavelengths it produces make combine_1d crash.

hbushouse commented on CHANGES.rst:
```diff
@@ -37,11 +37,18 @@ extract_1d
+extract_2d
+----------
+
+- Replaced a white space in the names of grism objects with an upderscore. [#3517]
```
Typo: "u**p**derscore"

nden commented:
The failure on Jenkins is unrelated and due to a change in `photutils`. it is being fixed in a separate PR.
